shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/ContextManager.java

```diff
@@ -63,6 +79,74 @@ public synchronized void renewTransactionContexts(final TransactionContexts tran
         this.transactionContexts = transactionContexts;
     }
     
+    /**
+     * Add schema.
+     * 
+     * @param schemaName schema name
+     * @throws SQLException SQL exception                  
+     */
+    public void addSchema(final String schemaName) throws SQLException {
+        if (metaDataContexts.getMetaDataMap().containsKey(schemaName)) {
+            return;
+        }
+        MetaDataContexts newMetaDataContexts = buildNewMetaDataContext(schemaName);
+        metaDataContexts.getOptimizerContext().getMetaData().getSchemas().put(schemaName,
+                newMetaDataContexts.getOptimizerContext().getMetaData().getSchemas().get(schemaName));
+        metaDataContexts.getMetaDataMap().put(schemaName, newMetaDataContexts.getMetaData(schemaName));
+    }
+    
+    /**
+     * Delete schema.
+     * 
+     * @param schemaName schema name
+     */
+    public void deleteSchema(final String schemaName) {
+        if (metaDataContexts.getMetaDataMap().containsKey(schemaName)) {
+            metaDataContexts.getOptimizerContext().getMetaData().getSchemas().remove(schemaName);
+            metaDataContexts.getOptimizerContext().getParserContexts().remove(schemaName);
+            metaDataContexts.getOptimizerContext().getPlannerContexts().remove(schemaName);
+            ShardingSphereMetaData removeMetaData = metaDataContexts.getMetaDataMap().remove(schemaName);
+            closeDataSources(removeMetaData);
+            closeTransactionEngine(schemaName);
+        }
+    }
+    
+    private MetaDataContexts buildNewMetaDataContext(final String schemaName) throws SQLException {
+        Map<String, Map<String, DataSource>> dataSourcesMap = Collections.singletonMap(schemaName, new HashMap<>());
```

Comment:
map inside a map ? I'm not sure if multimap is better here . not formaliar with that . 

multimap ... from guava.